brickstrap.sh

```diff
@@ -311,32 +394,26 @@ function configure-packages () {
     ${CHROOTCMD} ln -sf /usr/bin/gawk /usr/local/bin/awk
 
     # preseed debconf
-    info "preseed debconf"
-    if [ -r "${BOARDDIR}/debconfseed.txt" ]; then
-        cp "${BOARDDIR}/debconfseed.txt" "${ROOTDIR}/tmp/"
-        ${CHROOTCMD} debconf-set-selections /tmp/debconfseed.txt
-        rm "${ROOTDIR}/tmp/debconfseed.txt"
+    if br_list_paths debconfseed.txt -r >/dev/null; then
+        info "preseed debconf"
+        br_for_each_path "$(br_list_paths debconfseed.txt -r)" \
+            brp_preseed_debconf
     fi
 
     # run preinst scripts
     info "running preinst scripts..."
-    script_dir="${ROOTDIR}/var/lib/dpkg/info"
-    for script in ${script_dir}/*.preinst; do
-        blacklisted="false"
-        if [ -r "${BOARDDIR}/preinst.blacklist" ]; then
-            while read line; do
-                if [ "${script##$script_dir}" = "/${line}.preinst" ]; then
-                    blacklisted="true"
-                    info "skipping ${script##$script_dir} (blacklisted)"
-                    break
-                fi
-            done < "${BOARDDIR}/preinst.blacklist"
+    BRP_script_dir="${ROOTDIR}/var/lib/dpkg/info"
+    BRP_preinst_blacklist="$(br_cat_files preinst.blacklist)"
+    for BRP_script in ${BRP_script_dir}/*.preinst; do
+        if echo "$BRP_preinst_blacklist" | \
+            grep -q "$(basename "$BRP_script" .preinst)"; then
```

Comment:
Right, but `.` is a special regex char, so `foo1.0` would match `foo100` too. The odds are low that there are actually package names where this would happen are probably pretty low, but it *could* happen, especially with kernel packages that have long version strings in the package name.